### PR TITLE
PCVL-391 compose postselect

### PR DESCRIPTION
**Pull request by ericbrts**

### CODE REVIEW

File: perceval/utils/postselect.py [hunk 1]
```diff
@@ -130,6 +130,7 @@ def apply_permutation(self, perm_vector: List[int], first_mode: int = 0):
 
         :param perm_vector: Permutation vector to apply (as returned by PERM.perm_vector)
         :param first_mode: First mode of the permutation to apply (default 0)
+        :return: A PostSelect with the permutation applied
```
Comment (MarionQuandela) [inline perceval/utils/postselect.py:133]: 👍

**Review by MarionQuandela — CHANGES_REQUESTED** [review]

File: perceval/components/abstract_processor.py [hunk 2]
```diff
@@ -360,8 +366,7 @@ def linear_circuit(self, flatten: bool = False) -> Circuit:
         Creates a linear circuit from internal components, if all internal components are unitary.
         :param flatten: if True, the component recursive hierarchy is discarded, making the output circuit "flat".
         """
-        if not self._is_unitary:
-            raise RuntimeError("Cannot retrieve a linear circuit because some components are non-unitary")
+        assert self._is_unitary, "Cannot retrieve a linear circuit because some components are non-unitary"
```
[the remark this answers is not in the document]
Comment (ericbrts) [reply perceval/components/abstract_processor.py:369]: Done

File: perceval/components/abstract_processor.py [hunk 3]
```diff
@@ -246,10 +242,19 @@ def add(self, mode_mapping, component, keep_port=True):
...
+        if self._postselect is not None and isinstance(self._postselect, PostSelect):
+            impacted_modes = list(mode_mapping.keys())
+            assert self._postselect.can_compose_with(impacted_modes),\
+                f"Post-selection conditions cannot compose with modes {impacted_modes}"
+
+    def _compose_processor(self, connector: ModeConnector, processor, keep_port: bool):
         self._is_unitary = self._is_unitary and processor._is_unitary
         self._has_td = self._has_td or processor._has_td
         mode_mapping = connector.resolve()
+        assert self._postselect is None or processor._postselect is None, \
```
[the remark this answers is not in the document]
Comment (ericbrts) [reply perceval/components/abstract_processor.py:255]: Done

File: perceval/components/_mode_connector.py [hunk 4]
```diff
@@ -117,7 +117,7 @@ def resolve(self):
             return self._map
 
         # Handle list input case
```
[the remark this answers is not in the document]
Comment (ericbrts) [reply perceval/components/_mode_connector.py:119]: Done

**Review by MarionQuandela — APPROVED** [review]

**Pull request merged**